rs/src/management/http_client.rs

```diff
@@ -234,21 +234,22 @@ impl TunnelManagementClient {
         self.execute_json("create_tunnel_port", request).await
     }
 
-    /// Updates an existing port on the tunnel.
+    /// Deletes an existing port on the tunnel.
     pub async fn delete_tunnel_port(
         &self,
         locator: &TunnelLocator,
         port_number: u16,
         options: &TunnelRequestOptions,
-    ) -> HttpResult<TunnelPort> {
+    ) -> HttpResult<()> {
         let url = self.build_tunnel_uri(
             locator,
             Some(&format!("{}/{}", PORTS_API_SUB_PATH, port_number)),
         );
         let request = self
             .make_tunnel_request(Method::DELETE, url, options)
             .await?;
-        self.execute_json("create_tunnel_port", request).await
+        self.execute_json::<NeverDeserialize>("delete_tunnel_port", request).await?;
```

Comment:
The service returns 404 from a request to delete a port that doesn't exist. In the other SDKs that 404 is converted to a `false` return value. Does it make sense to do the same here? Also for `delete_tunnel()`.

Reply:
This will return an HTTP error if the service gives back a 404